pkg/plugins/optional/helm/v2alpha/scaffolds/internal/kustomize/helm_templater.go

```diff
@@ -263,27 +363,298 @@ func (t *HelmTemplater) templateVolumes(yamlContent string) string {
+			}
+		case strings.Contains(trimmed, "--health-probe-bind-address"):
+			healthLine = line
+		case strings.Contains(trimmed, "--webhook-cert-path"),
+			strings.Contains(trimmed, "--metrics-cert-path"):
+			preservedLines = append(preservedLines, line)
+		default:
+			// Remaining args will be handled through values.yaml
+		}
+	}
+
+	var builder strings.Builder
+	builder.WriteString(indent)
+	builder.WriteString("args:\n")
+
+	if metricsLine != "" {
+		if metricsIndent == "" {
+			metricsIndent = itemIndent
+		}
+		builder.WriteString(metricsIndent)
+		builder.WriteString("{{- if .Values.metrics.enable }}\n")
+		builder.WriteString(metricsLine)
+		builder.WriteString("\n")
+		builder.WriteString(metricsIndent)
+		builder.WriteString("{{- else }}\n")
+		builder.WriteString(metricsIndent)
+		builder.WriteString("# Bind to :0 to disable the controller-runtime managed metrics server\n")
+		builder.WriteString(metricsIndent)
+		builder.WriteString("- --metrics-bind-address=:0\n")
+		builder.WriteString(metricsIndent)
+		builder.WriteString("{{- end }}\n")
+	}
+	if healthLine != "" {
+		builder.WriteString(healthLine)
+		builder.WriteString("\n")
+	}
+
+	builder.WriteString(itemIndent)
+	builder.WriteString("{{- range .Values.controllerManager.args }}\n")
+	builder.WriteString(itemIndent)
+	builder.WriteString("- {{ . }}\n")
+	builder.WriteString(itemIndent)
+	builder.WriteString("{{- end }}\n")
+
+	for _, line := range preservedLines {
+		builder.WriteString(line)
+		builder.WriteString("\n")
+	}
+
+	newBlock := strings.TrimRight(builder.String(), "\n") + "\n"
+
+	return yamlContent[:loc[0]] + newBlock + yamlContent[loc[1]:]
+}
+
 // templateImageReference converts hardcoded image references to Helm templates
 func (t *HelmTemplater) templateImageReference(yamlContent string) string {
-	// Replace hardcoded controller image with Helm template
-	// This handles the common case where kustomize outputs "controller:latest"
-	// or other hardcoded image references
-	imagePattern := regexp.MustCompile(`(\s+)image:\s+controller:latest`)
-	yamlContent = imagePattern.ReplaceAllString(yamlContent,
-		`${1}image: "{{ .Values.controllerManager.image.repository }}:{{ .Values.controllerManager.image.tag }}"`)
-
-	// Also handle any other common image patterns that might appear
-	imagePattern2 := regexp.MustCompile(`(\s+)image:\s+([^"'\s]+):(latest|[\w\.\-]+)`)
-	yamlContent = imagePattern2.ReplaceAllStringFunc(yamlContent, func(match string) string {
-		// Only replace if it looks like a controller image (contains "controller" or "manager")
-		if strings.Contains(match, "controller") || strings.Contains(match, "manager") {
-			indentMatch := regexp.MustCompile(`^(\s+)`)
-			indent := indentMatch.FindString(match)
-			return fmt.Sprintf(
-				`%simage: "{{ .Values.controllerManager.image.repository }}:{{ .Values.controllerManager.image.tag }}"`, indent)
-		}
-		return match
-	})
+	if !strings.Contains(yamlContent, "name: manager") {
```

Comment:
Just to confirm, this means that we _only_ allow the `values.yaml` to be used to specify the `manager` container image. If there are any sidecar/init containers, etc, we will keep the image as templated in Kustomize.
Is that expected?